internal/services/containers/kubernetes_fleet_manager_resource.go

```diff
@@ -88,30 +108,33 @@ func (r KubernetesFleetManagerResource) Create() sdk.ResourceFunc {
 		Timeout: 30 * time.Minute,
 		Func: func(ctx context.Context, metadata sdk.ResourceMetaData) error {
 			client := metadata.Client.ContainerService.V20240401.Fleets
+			subscriptionId := metadata.Client.Account.SubscriptionId
 
-			var config KubernetesFleetManagerResourceSchema
-			if err := metadata.Decode(&config); err != nil {
+			var model KubernetesFleetManagerModel
+			if err := metadata.Decode(&model); err != nil {
 				return fmt.Errorf("decoding: %+v", err)
 			}
 
-			subscriptionId := metadata.Client.Account.SubscriptionId
-
-			id := fleets.NewFleetID(subscriptionId, config.ResourceGroupName, config.Name)
+			id := fleets.NewFleetID(subscriptionId, model.ResourceGroupName, model.Name)
 
 			existing, err := client.Get(ctx, id)
-			if err != nil {
-				if !response.WasNotFound(existing.HttpResponse) {
-					return fmt.Errorf("checking for the presence of an existing %s: %+v", id, err)
-				}
+			if err != nil && !response.WasNotFound(existing.HttpResponse) {
+				return fmt.Errorf("checking for existing %s: %+v", id, err)
```

Comment:
Nit: This is a boilerplate error message, for consistency with the rest of the provider we should keep it as it was (there are some older resources that use `checking for existing` but these should be updated over time)

```suggestion
				return fmt.Errorf("checking for presence of existing %s: %+v", id, err)
```